src/backend/distributed/planner/distributed_planner.c

```diff
@@ -66,14 +66,34 @@ static uint64 NextPlanId = 1;
 /* keep track of planner call stack levels */
 int PlannerLevel = 0;
 
+typedef struct DistributedPlanningContext
+{
+	/* The parsed query that is given to the planner. It is a slightly modified
+	 * to work with the standard_planner */
+	Query *parse;
+
+	/* A copy of the original parsed query that is given to the planner. This
+	 * doesn't contain most of the changes that are made to parse. There's one
+	 * that change that is made for non fast path router queries though, which
+	 * is the assigning of RTE identities using AssignRTEIdentities. This is
+	 * NULL for non distributed plans, since those don't need it. */
+	Query *originalQuery;
+
+	/* the cursor options given to the planner */
+	int cursorOptions;
+
+	/* the ParamListInfo that is given to the planner */
+	ParamListInfo boundParams;
```

Comment:
I guess not,  but should we also add `hasUnresolvedParams` here? Do you have any comments on this?